Translators/WZDx/geospatial_service.py

```diff
@@ -88,16 +120,23 @@ def point_to_route_id(lon, lat):
         route (str): Route ID'''
     # colorado_extent = "-109.081667,37.002220,-102.028444,40.979583"
     # NOTE: currently failing to find proper routes on occasion...
-    r = requests.get(
-        f'{get_map_server_endpoint()}/identify?geometry={lon},{lat}&geometryType=esriGeometryPoint&sr=4326&tolerance=50&mapExtent={get_ten_meter_extent(lon,lat)}&imageDisplay=600,550,96&returnGeometry=false&returnZ=false&returnM=false&returnUnformattedValues=false&returnFieldName=false&f=json', verify=False)
-    data = r.json()
-    if (len(data['results']) > 1):
-        str_results = [*map(lambda x: x['attributes']
-                            ['RouteId_Legacy'], data['results'])]
-        print('Multiple routes found: ' + ', '.join(str_results))
-    if data["results"] == []:
-        return None
-    return data["results"][0]["attributes"]["RouteId_Legacy"]
+    global cache
+    key = f'point_to_route_id:{lon}:{lat}'
+    cache_val = check_cache(key)
+    if cache_val != None:
+        return cache_val
+    else:
+        r = requests.get(
+            f'{get_map_server_endpoint()}/identify?geometry={lon},{lat}&geometryType=esriGeometryPoint&sr=4326&tolerance=50&mapExtent={get_ten_meter_extent(lon,lat)}&imageDisplay=600,550,96&returnGeometry=false&returnZ=false&returnM=false&returnUnformattedValues=false&returnFieldName=false&f=json', verify=False)
+        data = r.json()
+        if (len(data['results']) > 1):
+            str_results = [*map(lambda x: x['attributes']
+                                ['RouteId_Legacy'], data['results'])]
+            print('Multiple routes found: ' + ', '.join(str_results))

```

Comment:
Let's tweak this one for logging as well.

Reply:
Updated to logging.info